server/rooms.ts

```diff
@@ -668,44 +668,42 @@ export abstract class BasicRoom {
 
 	/**
 	 * @param newID Add this param if the roomid is different from `toID(newTitle)`
+	 * @param moveLogs Whether or not to rename the log file. Defaults to true.
+	 * @param keepAliases Whether or not to point the room's aliases and name to its new name. Defaults to true.
 	 */
-	async rename(newTitle: string, newID?: RoomID) {
+	async rename(newTitle: string, newID?: RoomID, moveLogs = true, keepAliases = true) {
 		if (!newID) newID = toID(newTitle) as RoomID;
-		if (this.game || this.tour) return;
+		if ((this.type === 'battle' && moveLogs) || (this.type === 'chat' && this.game) || this.tour) return;
```

Comment:
`Chat.ErrorMessage`, right?

```suggestion
		if (this.type === 'chat' && this.game) {
			throw new Chat.ErrorMessage(`Please finish your game (${this.game.title}) before renaming ${this.roomid}.`);
		}
		if (this.type === 'battle' && moveLogs) {
			idk, think of a suitable error message
		}
```